File: P1/E3/Organizador.java
package E3;

class Organizador implements Empleado{

    private EventoBuilder builder;
    private String nombreOrg;

    public Organizador(String nombreOrg) {
        this.nombreOrg = nombreOrg;
    }

    public void construirEvento(String nombre, String fecha, String ubicacion) {
        builder.construirNombre(nombre);
        builder.construirFecha(fecha);
        builder.construirUbicacion(ubicacion);
    }

    public Evento getEvento() {
        return builder.getEvento();
    }

    @Override
    public String getNombre(){
        return ("Organizador: " + nombreOrg);
    }

    public void setBuilder(EventoBuilder builder) {
        this.builder = builder;
    }
}

File: P4/lib/organizador.dart
import 'package:flutter/material.dart';

import 'evento_builder.dart';
import 'evento.dart';
import 'empleado.dart';

class Organizador implements Empleado {
  late EventoBuilder _builder;
  late String _nombreOrg;
  late bool tieneEvento = false;
  late String nombreJefe;

  Organizador(String nombreOrg) {
    _nombreOrg = nombreOrg;
  }

  void construirEvento(String nombre, String fecha, String ubicacion) {
    //SOLO PUEDE CONSTRUIR UN EVENTO
    if(!tieneEvento) {
      _builder.construirNombre(nombre);
      _builder.construirFecha(fecha);
      _builder.construirUbicacion(ubicacion);
      tieneEvento = true;
    }
  }

  //AÑADIDA
  void cambiarNombreEvento(String nombre) {
    _builder.construirNombre(nombre);
  }

  //AÑADIDA
  void cambiarFechaEvento(String fecha) {
    _builder.construirFecha(fecha);
  }

  //AÑADIDA
  void cambiarUbiEvento(String ubicacion) {
    _builder.construirUbicacion(ubicacion);
  }

  //AÑADIDA.
  EventoBuilder getBuilder() {
    return _builder;
  }

  Evento getEvento() {
    return _builder.evento;
  }

  @override
  String getNombre() {
    return "Organizador - $_nombreOrg";
  }

  void setBuilder(EventoBuilder builder) {
    _builder = builder;
  }
}
